aten/src/ATen/core/dispatch/OpSchema.h

```diff
@@ -127,12 +143,25 @@ struct ivalue_to_arg_type<ArrayRef<T>> {
   }
 };
 
-template<class ReturnType, class ParamTypes, class FuncType, FuncType* kernel> struct _wrapKernel {};
-template<class ReturnType, class... ParamTypes, class FuncType, FuncType* kernel> struct _wrapKernel<ReturnType, guts::typelist::typelist<ParamTypes...>, FuncType, kernel> {
+template<class ReturnType, class ParamTypes, class StateType, class FuncType, FuncType* kernel> struct _wrapKernel {};
+template<class ReturnType, class... ParamTypes, class StateType, class FuncType, FuncType* kernel> struct _wrapKernel<ReturnType, guts::typelist::typelist<ParamTypes...>, StateType, FuncType, kernel> {
```

Comment:
The changes here are clear as mud to me. Mumble mumble new StateType parameter, mumble mumble copy paste?

Reply:
Yes, it's basically just changing the wrapper so that the kernel function (and the wrapper itself) get an additional `c10::KernelState*` parameter.